scripts/brave_rewards/publisher/twitter/twitter.ts

```
@@ -0,0 +1,559 @@
+
+let registeredOnUpdatedTab = false
+
+let configureTipActionTimeout: any = null
+
+let credentialHeaders = {}
+
+let newTwitter = true
+
+interface MediaMetaData {
+  user: {
+    id: string
+    screenName: string
+    fullName: string
+  }
+  post: {
+    id: string
+    timestamp: string
+    text: string
+  }
+}
+
+// FIXME: This needs to use localized strings
+const getMessage = (id: string) => {
+  switch (id) {
+    case 'twitterTipsHoverText':
+      return 'Tip this tweet'
+    case 'twitterTipsIconLabel':
+      return 'Tip'
+  }
+
+  return ''
+}
+
+const sendAPIRequest = (name: string, url: string) => {
+  return new Promise((resolve, reject) => {
+    if (!name || !url) {
+      reject(new Error('Invalid parameters'))
+      return
+    }
+
+    if (Object.keys(credentialHeaders).length === 0) {
+      reject(new Error('Missing credential headers'))
+      return
+    }
+
+    if (!port) {
+      reject(new Error('Invalid port'))
+      return
+    }
+
+    port.postMessage({
+      type: 'MakeAPIRequest',
+      mediaType: types.mediaType,
+      data: {
+        name,
+        url,
+        init: {
+          credentials: 'include',
+          headers: {
+            ...credentialHeaders
+          },
+          referrerPolicy: 'no-referrer-when-downgrade',
+          method: 'GET',
+          redirect: 'follow'
+        }
+      }})
+
+    port.onMessage.addListener(function (msg) {
+      if (!msg || !msg.data) {
+        reject(new Error('Invalid message'))
+        return
+      }
+      if (msg.type === 'MakeAPIRequest') {
+        if (!msg.data.name || !msg.data.response) {
+          reject(new Error('Invalid message'))
+          return
+        }
+        if (msg.data.name === name) {
+          resolve(msg.data.response)
+          return
+        }
+      }
+    })
+  })
+}
+
+const getTweetDetails = async (tweetId: string) => {
+  if (!tweetId) {
+    return Promise.reject('Invalid parameters')
```

We should reject with an error: `new Error('Invalid parameters')`. 

Since this is an async function, you can also just do `throw new Error('Invalid parameters')`.

---

Thanks, fixed.